### PR TITLE
QW-2126: Remove metastore trigger and replace with helper function.

### PR DESCRIPTION
**Pull request by ChillFish8**
As discussed in #2126 the Postgres trigger seems to have a significant performance impact when deleting rows from the splits table due to it running another `UPDATE` operation on the `indexes` table for every row that gets deleted.

This removes the trigger and instead has a helper function that gets ran when ever we call a method that modifies the splits table (this is not done implicitly though)
another potential solution was changing from `EACH ROW` to `EACH STATEMENT` but unfortunately, we lose the context of the query itself so cannot use this to update the index efficient and implicitly.

### CODE REVIEW

File: quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs [hunk 1]
```diff
@@ -172,10 +172,32 @@ async fn mark_splits_as_published_helper(
     .map(|row| row.get(0))
     .fetch_all(tx)
     .await?;
+
```
Comment (guilload) [inline quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:175]: That's just noise.

File: quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs [hunk 2]
```diff
@@ -172,10 +172,32 @@ async fn mark_splits_as_published_helper(
...
     Ok(published_split_ids)
 }
 
-/// Marks mutiple splits for deletion.
+#[instrument(skip(tx))]
+/// Updates the given index's `update_timestamp` to the current
+/// timestamp.
+async fn update_index_last_update_ts(
```
Comment (guilload) [inline quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:182]: `touch_index`?

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs (hunk 2 above)
Comment (ChillFish8) [reply quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:182]: I think `touch_index` is a bit vague about what it actually does, but I can't think of a better name for it :/

Comment (guilload) [reply quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:181]: Then, `update_index_update_timestamp` or `update_index_update_ts` is the most accurate.

**Review by guilload — APPROVED** [review]

**Pull request merged**